include/deal.II/lac/trilinos_sparse_matrix.h

```diff
@@ -2223,10 +2225,33 @@ namespace TrilinosWrappers
         * inverse operations.
         * The <tt>vmult</tt> and <tt>Tvmult</tt> operations retain the standard
         * definitions inherited from @p op.
+        *
+        * @note This function is enabled only if the solver and preconditioner
+        * derive from the respective TrilinosWrappers base classes.
         */
         template <typename Solver, typename Preconditioner>
-        TrilinosPayload inverse_payload (Solver &, const Preconditioner &) const;
+        typename std::enable_if<
+        std::is_base_of<TrilinosWrappers::SolverBase,Solver>::value and
```

Comment:
I didn't mean to shut the door on this. I don't think we've been doing this in the past, and it warrants a discussion :-)

My personal opinion, though, is to stick with `&&`, `||`, and `!`. I find that easier to read because the text form of these operators does not stand out as much from variable and class names. But that's just my opinion.